### PR TITLE
feat: migrate store/returns to typescript

### LINKED ISSUE #728: Convert to typescript -store/returns

### PR DESCRIPTION
**Pull request by zakariaelas**
fixes #728

### CODE REVIEW

**Review by olivermrbl — APPROVED** [review]

File: packages/medusa-js/src/resources/returns.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import BaseResource from './base';
-import * as Types from '../types';
+import { AxiosPromise } from 'axios'
```
Comment (olivermrbl) [inline packages/medusa-js/src/resources/returns.ts:2]: nit: Missing linting

File: packages/medusa/src/api/routes/store/returns/create-return.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
-import { MedusaError, Validator } from "medusa-core-utils"
+import { Type } from 'class-transformer';
+import { IsArray, IsNotEmpty, IsNumber, IsOptional, IsString, Min, ValidateNested } from 'class-validator';
+import { MedusaError } from "medusa-core-utils";
+import EventBusService from '../../../../services/event-bus';
+import IdempotencyKeyService from '../../../../services/idempotency-key';
+import OrderService from '../../../../services/order';
+import ReturnService from '../../../../services/return';
+import { validator } from "../../../../utils/validator";
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/returns/create-return.ts:8]: Two different types of lintin here

File: packages/medusa-js/src/resources/returns.ts (hunk 1 above)
Comment (zakariaelas) [reply packages/medusa-js/src/resources/returns.ts:2]: can we try to enforce this through a linting rule? Hoping we could use something like [this](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md). What do you think?

Comment (olivermrbl) [reply packages/medusa-js/src/resources/returns.ts:2]: One of the other PRs include a fix, that will ensure correct linting of the client :) It is currently a `prettier` file in the client folder, that tampers with out `eslint`.

**Pull request merged**